core/embed/extmod/modtrezormintlayer/modtrezormintlayer.h

```diff
@@ -0,0 +1,693 @@
+#include <stdio.h>
+#include "py/objstr.h"
+
+#include "embed/extmod/trezorobj.h"
+
+#include "embed/rust/mintlayer.h"
+
+#include "bip32.h"
+#include "bip39.h"
+#include "curves.h"
+#include "memzero.h"
+
+void handle_err(ByteArray *res) {
+  if (res->data != NULL) {
+    return;
+  }
+
+  switch (res->len_or_err.err) {
+    case WrongHashSize:
+      mp_raise_ValueError("Invalid hash size");
+      break;
+    case InvalidUtxoType:
+      mp_raise_ValueError("Invalid UTXO type");
+      break;
+    case InvalidAmount:
+      mp_raise_ValueError("Invalid amount");
+      break;
+    case InvalidAccountCommand:
+      mp_raise_ValueError("Invalid account command");
+      break;
+    case InvalidDestination:
+      mp_raise_ValueError("Invalid destination");
+      break;
+    case InvalidIsTokenUnfreezable:
+      mp_raise_ValueError("Invalid token unfreezable flag");
+      break;
+    case InvalidIsTokenFreezable:
+      mp_raise_ValueError("Invalid token freezable flag");
+      break;
+    case InvalidVrfPublicKey:
+      mp_raise_ValueError("Invalid VRF public key");
+      break;
+    case InvalidPublicKey:
+      mp_raise_ValueError("Invalid public key");
+      break;
+    case InvalidOutputTimeLock:
+      mp_raise_ValueError("Invalid output time lock");
+      break;
+    case InvalidTokenTotalSupply:
+      mp_raise_ValueError("Invalid token total supply");
+      break;
+    default:
+      mp_raise_ValueError("Unknown error");
+      break;
+  }
+}
+
+/// def encode_utxo_input(tx_hash: bytes, index: int, utxo_type: int) -> bytes:
+///     """
+///     encodes an utxo input from tx_hash and index
+///     """
+STATIC mp_obj_t mod_trezormintlayer_utils_mintlayer_encode_utxo_input(
+    mp_obj_t tx_hash, mp_obj_t index, mp_obj_t utxo_type_obj) {
+  mp_buffer_info_t hash = {0};
+  mp_get_buffer_raise(tx_hash, &hash, MP_BUFFER_READ);
+  if (hash.len != 32) {
+    printf("invalid hash len: %ld", (long int)hash.len);
+    mp_raise_ValueError("Invalid hash");
+  }
+  uint32_t idx = trezor_obj_get_uint(index);
+  uint32_t utxo_type = trezor_obj_get_uint(utxo_type_obj);
+  ByteArray arr =
+      mintlayer_encode_utxo_input(hash.buf, hash.len, idx, utxo_type);
+  handle_err(&arr);
+
+  vstr_t pkh = {0};
+  vstr_init_len(&pkh, arr.len_or_err.len);
+  int i = 0;
+  for (; i < arr.len_or_err.len; i++) {
+    ((uint8_t *)pkh.buf)[i] = (uint8_t)arr.data[i];
+  }
```

Comment:
1) Plz use a miningful name instead of "pkh" (it's not a public key hash here).
2) Use `memcpy` instead of looping.

Same in other places.